arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -448,6 +473,14 @@ public String toString() {
 
     @Override
     public void setBatchSize(int batchSize) {
+      this.batchSize = (batchSize == 0) ? DEFAULT_BATCH_SIZE : batchSize;
```

Comment:
I think this needs to reallocate the nullability holder if batch size increased, or else a batch size change could cause it to overflow.